VocaDb.ResXFileCodeGenerator/StringBuilderGenerator.cs

```diff
@@ -1,140 +1,207 @@
-	private static readonly Regex InvalidMemberNameSymbols = new(@"[^\p{Cf}\p{L}\p{Mc}\p{Mn}\p{Nd}\p{Nl}\p{Pc}]", RegexOptions.Compiled | RegexOptions.CultureInvariant);
-	public string Generate(Stream resxStream, GeneratorOptions options)
+	private static readonly Regex s_validMemberNamePattern =
+		new(@"^[\p{L}\p{Nl}_][\p{Cf}\p{L}\p{Mc}\p{Mn}\p{Nd}\p{Nl}\p{Pc}]*$",
+			RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+	private static readonly Regex s_invalidMemberNameSymbols = new(@"[^\p{Cf}\p{L}\p{Mc}\p{Mn}\p{Nd}\p{Nl}\p{Pc}]",
+		RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+	private static readonly DiagnosticDescriptor s_duplicateWarning = new("VocaDbResXFileCodeGenerator001",
+		"Duplicate member",
+		"Ignored added member '{0}'",
+		"ResXFileCodeGenerator",
+		DiagnosticSeverity.Warning,
+		true);
+
+	private static readonly DiagnosticDescriptor s_memberSameAsClassWarning = new("VocaDbResXFileCodeGenerator002",
+		"Member same name as class",
+		"Ignored member '{0}' has same name as class",
+		"ResXFileCodeGenerator",
+		DiagnosticSeverity.Warning,
+		true);
+
+	private static readonly DiagnosticDescriptor s_memberWithStaticError = new("VocaDbResXFileCodeGenerator003",
+		"Incompatible settings",
+		"Cannot have static members/class with an class instance",
+		"ResXFileCodeGenerator",
+		DiagnosticSeverity.Error,
+		true);
+
+	public (string generatedFileName, string SourceCode, IEnumerable<Diagnostic> ErrorsAndWarnings) Generate(
+		FileOptions options, CancellationToken cancellationToken = default)
 	{
-		// HACK: netstandard2.0 doesn't support improved interpolated strings?
-		var builder = new StringBuilder();
-
-		builder.AppendLine(Constants.AutoGeneratedHeader);
-		builder.AppendLine("#nullable enable");
-
-		builder.Append("namespace ");
-		builder.AppendLine(options.CustomToolNamespace ?? options.LocalNamespace);
-		builder.AppendLine("{");
+		var errorsAndWarnings = new List<Diagnostic>();
+		var generatedFileName = $"{options.LocalNamespace}.{options.ClassName}.g.cs";
 
-		builder.Append("    using ");
-		builder.Append(Constants.SystemGlobalization);
-		builder.AppendLine(";");
+		var content = options.File.GetText(cancellationToken);
+		if (content is null) return (generatedFileName, "//ERROR reading file:" + options.File.Path, errorsAndWarnings);
 
-		builder.Append("    using ");
-		builder.Append(Constants.SystemResources);
-		builder.AppendLine(";");
+		// HACK: netstandard2.0 doesn't support improved interpolated strings?
+		var builder = GetBuilder(options.CustomToolNamespace ?? options.LocalNamespace);
 
-		builder.AppendLine();
+		if (options.UseVocaDbResManager)
+			AppendCodeUsings(builder);
+		else
+			AppendResourceManagerUsings(builder);
 
-		builder.Append("    ");
 		builder.Append(options.PublicClass ? "public" : "internal");
+		builder.Append(options.PartialClass ? " partial" : "");
 		builder.Append(options.StaticClass ? " static class " : " class ");
```

Comment:
The `partial` modifier can only appear immediately before `class`.

Reply:
Will fix